docs/specification/draft/client/elicitation.mdx

```diff
@@ -321,12 +526,212 @@ Servers should handle each state appropriately:
+
+1. Servers **MUST NOT** include URLs in any message or schema fields as part of a form elicitation request.
+2. Servers **MUST NOT** include URLs in any message or schema fields as part of an url elicitation request, except for the `url` field.
+
+These requirements ensure that client implementations have clear rules about when to present a URL to the user, so that other rules about user consent and SSRF protection (below) can be consistently applied.
+
+### Identifying the User
+
+Servers **MUST NOT** rely on client-provided user identification, as this can be forged.
+Instead, servers **SHOULD** follow [security best practices](../basic/security_best_practices).
+
+Non-normative examples:
+
+- Incorrect: Treat user input like "I am joe@example.com" as authoritative
+- Correct: Rely on the [MCP authorization server](../basic/authorization) to identify the user
+
+### Form Mode Security
+
+1. Servers **MUST NOT** request sensitive information (passwords, API keys, etc.) via form mode
+2. Clients **SHOULD** validate all responses against the provided schema
+3. Servers **SHOULD** validate received data matches the requested schema
+
+### URL Mode Security
+
+Clients **MUST NOT** open a user agent (e.g. browser) without explicit consent from the user.
+
+Clients **MUST** render the URL provided by the server in a browser which does not enable the client or LLM to inspect the content or user inputs.
+For example, on iOS, [SFSafariViewController](https://developer.apple.com/documentation/safariservices/sfsafariviewcontroller) is good, but [WkWebView](https://developer.apple.com/documentation/webkit/wkwebview) is not.
+
+#### Server-Side Request Forgery (SSRF)
+
+Since clients facilitate the opening of URLs provided by servers, they **MUST** implement SSRF protections.
+
+The following are recommendations for implementing SSRF protections. Servers that choose to relax these constraints should be aware of the risks.
+
+- Require the `https://` scheme for all url mode URLs (no HTTP, `file://`, etc.)
+- Block requests to internal IP ranges (e.g., `127.0.0.1`, `10.0.0.0/8`, etc.)
+- Clearly render or distinguish Unicode characters (e.g. punycode URLs) to avoid "look-alike" misdirections
+- Clearly communicate the destination server and target URL to the user when asking for consent before opening the URL
+
+Further recommendations can be found in the OWASP [SSRF Prevention Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Server_Side_Request_Forgery_Prevention_Cheat_Sheet.html).
+
+#### Phishing
+
+URL elicitation returns a URL that an attacker can use to send to a victim. The MCP Server **MUST** verify the identity of the user who opens the URL before accepting information.
+
+Typically identity verification is done by leveraging the [MCP authorization server](../basic/authorization) to identify the user, through a session cookie or equivalent in the browser.
```

Comment:
I am not entirely clear how session cookies come into the mix here. This seems like a browser-specific construct when clients themselves might not be in a browser.

Reply:
It's an example of how the identity might be determined. Doesn't need to be a cookie, but it's a common example when URLs are opened in a browser.